dist/boards_local_txt.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+
+# adapted boards.txt.py from esp8266/Arduino
+# - single board definition, ldscripts
+# - portable boards.local.txt for 2.3.0 and up
+
+import os
+import argparse
+import sys
+import collections
+
+# TODO: drop after platform.io supports python 3
+if sys.version < (3, 2):
+    import string
+    _format = string.Formatter().vformat
+    def format_map(tmpl, f_map):
+        return _format(tmpl, None, f_map)
+else:
+    def format_map(tmpl, f_map):
+        return tmpl.format_map(f_map)
+
+
+class VersionedSubstitution(collections.MutableMapping):
+    def __init__(self, substitutions, targets, *args, **kwargs):
+        self._targets = targets
+        self._version = None
+        self._store = substitutions.copy()
+        self.update(dict(*args, **kwargs))
+
+    def set_version(self, version):
+        self._version = version
+
+    def __getitem__(self, key):
+        if self._version in self._targets:
+            return self._store[key]
+        return key
+
+    def __setitem__(self, key, value):
+        self._store[key] = value
+
+    def __delitem__(self, key):
+        del self._store[key]
+
+    def __iter__(self):
+        return iter(self._store)
+
+    def __len__(self):
+        return len(self._store)
+
+
+BOARDS_LOCAL = {
+    "global": collections.OrderedDict(
+        [("menu.float_support", "scanf and printf float support")]
+    ),
+    "flash_size": collections.OrderedDict(
+        [
+            (".menu.{eesz}.1M", "1M (1 EEPROM Sector, no SPIFFS)"),
```

Comment:
This collides with Core's 1M (no SPIFFS) option (2.4.0 and up, because 2.3.0 does not have it)
Names need to be unique.

sadly, no way to hide other ones. https://github.com/arduino/Arduino/issues/6034 is undocumented way to hide boards, but not menu options :(